src/IdeaStatiCa.IntermediateModel/Extensions/ISIntermediateExtension.cs

```diff
@@ -0,0 +1,298 @@
+		{
+			switch (intermediateItem)
+			{
+				case SObject sObject:
+					return sObject.TakeElementProperty(property);
+				case SList sList:
+					return sList.TakeElementProperty(property);
+				case SPrimitive sPrimitive:
+					return sPrimitive.TakeElementProperty(property);
+				case SAttribute sAttribute:
+					return sAttribute.TakeElementProperty(property);
+				default: throw new InvalidOperationException($"Unknown type of intermediateItem {intermediateItem.GetType()}");
+			}
+		}
+
+
+		/// <summary>
+		/// Add Element Property
+		/// </summary>
+		/// <param name="intermediateItem"></param>
+		/// <param name="property"></param>
+		/// <exception cref="InvalidOperationException"></exception>
+		public static void AddElementProperty(this ISIntermediate intermediateItem, ISIntermediate property)
+		{
+			switch (intermediateItem)
+			{
+				case SObject sObject:
+					sObject.AddElementProperty(property);
+					break;
+				case SList sList:
+					sList.AddElementProperty(property);
+					break;
+				case SPrimitive sPrimitive:
+					sPrimitive.AddElementProperty(property);
+					break;
+				case SAttribute sAttribute:
+					sAttribute.AddElementProperty(property);
+					break;
+				default: throw new InvalidOperationException($"Unsupported type of intermediateItem {intermediateItem.GetType()}");
+			}
+		}
+
+		/// <summary>
+		/// Try Add Element Property
+		/// </summary>
+		/// <param name="intermediateItem"></param>
+		/// <param name="property"></param>
+		/// <returns></returns>
+		/// <exception cref="InvalidOperationException"></exception>
+		public static bool TryAddElementProperty(this ISIntermediate intermediateItem, ISIntermediate property)
+		{
+			switch (intermediateItem)
+			{
+				case SObject sObject:
+					return sObject.TryAddElementProperty(property);
+				case SList sList:
+					return sList.TryAddElementProperty(property);
+				case SPrimitive sPrimitive:
+					return sPrimitive.TryAddElementProperty(property);
+				case SAttribute sAttribute:
+					return sAttribute.TryAddElementProperty(property);
+				default: throw new InvalidOperationException($"Unsupported type of intermediateItem {intermediateItem.GetType()}");
+			}
+		}
+
+
+		private static Queue<string> PrepareFilter(string filter)
+		{
+			var filterParts = ExplodeFilter(filter);
+			var queue = new Queue<string>();
+			filterParts.ForEach(fp => queue.Enqueue(fp));
+
+			return queue;
+		}
+
+		private static List<string> ExplodeFilter(string filterParh)
+		{
+			return filterParh.Split(";").ToList();
+		}
+
+		/// <summary>
+		/// Create Element Property
+		/// </summary>
+		/// <param name="intermediateItem"></param>
+		/// <param name="name"></param>
+		/// <returns></returns>
+		/// <exception cref="InvalidOperationException"></exception>
+		public static SObject CreateElementProperty(this ISIntermediate intermediateItem, string name)
+		{
+			switch (intermediateItem)
+			{
```

Comment:
Switch for one case seems bit too complicated

Reply:
rewrite to simple if